### PR TITLE
UX: Convert admin leaderboard forms to FormKit

### PR DESCRIPTION
**Pull request by martin-brennan**
Also use new AdminPageSubheader component on the
index page for the title and buttons.

image
image

### CODE REVIEW

**Comment by jjaffeux** [comment]
Screenshot 2024-09-11 at 10 55 57

nitpick and only UI related but this back buttons looks a little bit odd to me

File: admin/assets/javascripts/admin/components/admin-create-leaderboard.gjs [hunk 1]
```diff
@@ -20,62 +19,72 @@ export default class extends Component {
...
+            @showTitle={{false}}
+            class="new-leaderboard__name"
+            @validation="required"
+            as |field|
+          >
+            <field.Input
+              placeholder={{i18n "gamification.leaderboard.name_placeholder"}}
+            />
+          </form.Field>
+          <form.Submit />&nbsp;
```
Comment (jjaffeux) [inline admin/assets/javascripts/admin/components/admin-create-leaderboard.gjs:79]: ? :p

You have form actions for this use case:



```
<form.Actions>
    <form.Submit />
    
    {{#unless this.readOnly}}
      <form.Button
        @action={{this.handleDelete}}
        class="badge-form__delete-badge-btn btn-danger"
      >
        {{i18n "admin.badges.delete"}}
      </form.Button>
    {{/unless}}
</form.Actions>
```

File: admin/assets/javascripts/admin/components/admin-create-leaderboard.gjs [hunk 2]
```diff
@@ -20,62 +19,72 @@ export default class extends Component {
...
+          <DButton
```
Comment (jjaffeux) [inline admin/assets/javascripts/admin/components/admin-create-leaderboard.gjs:80]: and yes I would use form.Button here

File: admin/assets/javascripts/admin/components/admin-create-leaderboard.gjs (hunk 1 above)
Comment (martin-brennan) [reply admin/assets/javascripts/admin/components/admin-create-leaderboard.gjs:79]: Oh nice I didn't realise! Thanks

Comment (martin-brennan) [reply admin/assets/javascripts/admin/components/admin-create-leaderboard.gjs:79]: Where does this `readOnly` come from though, what's it for?

**Comment by martin-brennan** [comment]
> this back buttons looks a little bit odd to me

@jjaffeux this same one is used in a few places now, it's the `BackButton` component part of the new UI guidelines. Feel free to bring it up in `/t/129128` internally

**Review by tgxworld — APPROVED** [review]

**Pull request merged**